src/redis-check-rdb.c

```diff
@@ -186,17 +186,17 @@ void rdbCheckSetupSignals(void) {
     sigaction(SIGABRT, &act, NULL);
 }
 
-static int isFifo(char *filename) {
-    struct stat stat_p;
-    stat(filename, &stat_p);
-    return S_ISFIFO(stat_p.st_mode);
+static int isFifo(char *filepath) {
+    struct stat sb;
+    if (stat(filepath, &sb) == -1) return 0;
+    return S_ISFIFO(sb.st_mode);
 }
 
 /* Check the specified RDB file. Return 0 if the RDB looks sane, otherwise
  * 1 is returned.
- * The file is specified as a filename in 'rdbfilename' if 'fp' is not NULL,
+ * The file is specified as a filename in 'rdbfilepath' if 'fp' is NULL,
  * otherwise the already open file 'fp' is checked. */
-int redis_check_rdb(char *rdbfilename, FILE *fp) {
+int redis_check_rdb(char *rdbfilepath, FILE *fp) {
```

Comment:
btw, that rename is excessive IMHO, the old name was ok, and there's no need to mess up the blame log or cause unnecessary merge conflicts for that.